src/video/sdl_v.cpp

```diff
@@ -155,17 +155,22 @@ static void DrawSurfaceToScreen()
 	if (n == 0) return;
 
 	_num_dirty_rects = 0;
+
 	if (n > MAX_DIRTY_RECTS) {
-		if (_sdl_screen != _sdl_realscreen) {
-			SDL_BlitSurface(_sdl_screen, nullptr, _sdl_realscreen, nullptr);
+		if (_sdl_surface != _sdl_realscreen) {
+			SDL_BlitSurface(_sdl_surface, nullptr, _sdl_realscreen, nullptr);
 		}
+
 		SDL_UpdateRect(_sdl_realscreen, 0, 0, 0, 0);
 	} else {
-		if (_sdl_screen != _sdl_realscreen) {
+		if (_sdl_surface != _sdl_realscreen) {
 			for (int i = 0; i < n; i++) {
-				SDL_BlitSurface(_sdl_screen, &_dirty_rects[i], _sdl_realscreen, &_dirty_rects[i]);
+				SDL_BlitSurface(
```

Comment:
I'd change the line in SDL2, tbh. OTTD doesn't have line length limits, and this line isn't even that long

Reply:
Good idea; I am more than willing to do that. How do I edit a pull request?

Reply:
Never mind, I figured it out ;)